src/sdk/emitter/webAudioSoundEmitter.ts

```diff
@@ -8,6 +8,46 @@ import { SoundEmitter } from './soundEmitter';
  * @implements {SoundEmitter}
  */
 export class WebAudioSoundEmitter implements SoundEmitter {
+
+  /**
+   * Emits a sound through HTML5's AudioContext interface.
+   *
+   * @param {ArrayBuffer} sound A binary record of the sound you want to play.
+   * @returns {Promise<void>}
+   * @memberof WebAudioSoundEmitter
+   */
+
+
+  async emit(sound: ArrayBuffer): Promise<void> {
+    let context: AudioContext;
+
+    try {
+      context = new (window['AudioContext'] || window['webkitAudioContext'])();
+    } catch (e) {
+      console.error('AudioContext is not supported.');
+
+      throw e;
+    }
+    const source = context.createBufferSource();
+
+    context.decodeAudioData(sound, (decodedData) => {
+      source.buffer = decodedData;
+      source.connect(context.destination);
+      source.start(0);
+    },
+    (error) => {
+      console.error('Error with decoding audio data');
+      throw error;
```

Comment:
Throw sth from sdk/errors or add there a new error.

Reply:
IMO you don't have to pass error callback here at all, but if you do that - return sth from custom errors, but for me you can remove the callback